tools/celrpc/celrpc.go

```diff
@@ -0,0 +1,126 @@
+package celrpc
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+)
+
+type ConfClient struct {
+	exprpb.ConformanceServiceClient
+	cmd *exec.Cmd
+	conn *grpc.ClientConn
+}
+
+// NewClientFromPath creates a new ConformanceService gRPC client,
+// connecting to a server which is launched by the binary at the given
+// serverPath.  The spawned server shares the current process's stderr,
+// so its log messages will be visible.  The caller must call Shutdown()
+// on the retured ConfClient, even if NewClientFromPath() returns a
+// non-nil error.
+func NewClientFromPath(serverPath string) (*ConfClient, error) {
+	c := ConfClient{}
+
+	cmd := exec.Command(serverPath)
+	out, err := cmd.StdoutPipe()
+	if err != nil {
+		return &c, err
+	}
+	cmd.Stderr = os.Stderr // share our error stream
+
+	err = cmd.Start()
+	if err != nil {
+		return &c, err
+	}
+	// Only assign cmd for stopping if it has successfully started.
+	c.cmd = cmd
+
+	var addr string
+	_, err = fmt.Fscanf(out, "Listening on %s\n", &addr)
+	out.Close()
+	if err != nil {
+		return &c, err
+	}
+
+	conn, err := grpc.Dial(addr, grpc.WithInsecure())
+	if err != nil {
+		return &c, err
+	}
+	c.conn = conn
+
+	c.ConformanceServiceClient = exprpb.NewConformanceServiceClient(conn)
+	return &c, nil
+}
+
+func ExampleNewClientFromPath() {
```

Comment:
If you want this example to be executable, you could include it in a celrpc_test.go file with an output comment of the expected result:

```go
fmt.Println(...)
// Output: hello, world!
```

Reply:
Great idea - I'll do that in a followup.